arm-datalake-store/account/2016-11-01/swagger/account.json

```diff
@@ -0,0 +1,1321 @@
+            "modelAsString": false
+          }
+        },
+        "keyVaultMetaInfo": {
+          "$ref": "#/definitions/KeyVaultMetaInfo",
+          "description": "The Key vault information for connecting to user managed encryption keys."
+        }
+      },
+      "description": "The encryption configuration for the account."
+    },
+    "KeyVaultMetaInfo": {
+      "required": [
+        "keyVaultResourceId",
+        "encryptionKeyName",
+        "encryptionKeyVersion"
+      ],
+      "properties": {
+        "keyVaultResourceId": {
+          "type": "string",
+          "description": "The resource identifier for the user managed Key Vault being used to encrypt."
+        },
+        "encryptionKeyName": {
+          "type": "string",
+          "description": "The name of the user managed encryption key."
+        },
+        "encryptionKeyVersion": {
+          "type": "string",
+          "description": "The version of the user managed encryption key."
+        }
+      },
+      "description": "Metadata information used by account encryption."
+    },
+    "DataLakeStoreAccountProperties": {
+      "properties": {
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string",
+          "description": "the status of the Data Lake Store account while being provisioned.",
+          "enum": [
+            "Failed",
+            "Creating",
+            "Running",
+            "Succeeded",
+            "Patching",
+            "Suspending",
+            "Resuming",
+            "Deleting",
+            "Deleted"
+          ],
+          "x-ms-enum": {
+            "name": "DataLakeStoreAccountStatus",
+            "modelAsString": false
+          }
+        },
+        "state": {
+          "readOnly": true,
+          "type": "string",
+          "description": "the status of the Data Lake Store account after provisioning has completed.",
+          "enum": [
+            "active",
+            "suspended"
+          ],
+          "x-ms-enum": {
+            "name": "DataLakeStoreAccountState",
+            "modelAsString": false
+          }
+        },
+        "creationTime": {
+          "type": "string",
+          "readOnly": true,
+          "format": "date-time",
+          "description": "the account creation time."
+        },
+        "encryptionState": {
+          "type": "string",
+          "description": "The current state of encryption for this Data Lake store account.",
+          "enum": [
+            "Enabled",
+            "Disabled"
+          ],
+          "x-ms-enum": {
+            "name": "EncryptionState",
+            "modelAsString": false
+          }
+        },
+        "encryptionProvisioningState": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The current state of encryption provisioning for this Data Lake store account.",
+          "enum": [
+            "Creating",
+            "Succeeded"
```

Comment:
This is an implementation choice from our service. The "Creating" state is a the non-success state. So if encryption can't be enabled for some reason, it will stay in the "Creating" state. When calling "enableKeyVault" method, that is when you will see an error as to why the provisioning state can't move to succeeded.